### PR TITLE
[trace-view] Added support for native functions and global locations

### PR DESCRIPTION
**Pull request by awelc**
## Description 

Adds support for native functions where the `CloseFrame` event happens right after `OpenFrame` event and both need to be skipped to skip over the native function. Also added support for handling global locations - they should be parsed properly but largely ignored in the trace as they cannot directly affect values of local variables

## Test plan 

All old and new tests must pass

### CODE REVIEW

File: external-crates/move/crates/move-analyzer/trace-adapter/src/trace_utils.ts [hunk 1]
```diff
@@ -506,7 +516,9 @@ function processJSONLocalLocation(
     } else if ('Indexed' in traceLocation) {
         return processJSONLocalLocation(traceLocation.Indexed[0], localLifetimeEnds);
     } else {
-        throw new Error(`Unsupported location type: ${JSON.stringify(traceLocation)}`);
+        // there is nothing that needs to be done for 'Global' locations,
+        // neither with respect to lifetime nor with respect to location itself
```
Comment (tzakian) [inline external-crates/move/crates/move-analyzer/trace-adapter/src/trace_utils.ts:520]: Just want to note that this is not exactly true, but for all intents and purposes this is fine. If we ever have a native function that takes a mutable reference and returns a field of that struct by mutable reference something would need to be done to support that here. However, I don't such a function will ever exist.

**Review by tzakian — APPROVED** [review]
LGTM! I'm somewhat sad that the sources need to be copied over, but I don't have a better suggestion on how to support it other than what you have right now (but just something to think about for future work).

File: external-crates/move/crates/move-analyzer/trace-adapter/src/trace_utils.ts (hunk 1 above)
Comment (awelc) [reply external-crates/move/crates/move-analyzer/trace-adapter/src/trace_utils.ts:520]: I did have that in mind when writing this code, but I should still add a comment there along these lines for posterity

**Comment by awelc** [comment]
> LGTM! I'm somewhat sad that the sources need to be copied over, but I don't have a better suggestion on how to support it other than what you have right now (but just something to think about for future work).

Yeah, I don't love it either, but since the blast radius is somewhat limited, I wasn't worrying about it too much

**Pull request merged**